test/test_bson.py

```diff
@@ -530,7 +530,9 @@ def test_large_datetime_truncation(self):
     def test_aware_datetime(self):
         aware = datetime.datetime(1993, 4, 4, 2,
                                   tzinfo=FixedOffset(555, "SomeZone"))
-        as_utc = (aware - aware.utcoffset()).replace(tzinfo=utc)
+        offset = aware.utcoffset()
+        assert offset is not None
+        as_utc = (aware - offset).replace(tzinfo=utc)
```

Comment:
This is disappointing. IIUC a datetime with a tzinfo should always return a non-None utcoffset(). I think it could be related to https://github.com/python/mypy/issues/10067 but I think it's worth reporting as a separate issue.

Reply:
I think the only way to do it is to do what I suggested in https://github.com/python/mypy/issues/10067#issuecomment-1032041938